### PR TITLE
refactor various detectors to use RegexBasedDetector

### PR DESCRIPTION
**Pull request by JoshuaRLi**
Follow up to the `RegexBasedDetector` implemented in https://github.com/Yelp/detect-secrets/pull/100.

### CODE REVIEW

**Comment by KevinHock** [comment]
Let's merge in https://github.com/Yelp/detect-secrets/pull/86 before this one

**Comment by JoshuaRLi** [comment]
If that's done, then the changes to `KeywordDetector` in this branch should be discarded.

File: detect_secrets/plugins/base.py [hunk 1]
```diff
@@ -126,5 +126,5 @@ def analyze_string(self, string, line_num, filename):
 
     def secret_generator(self, string):
         for regex in self.blacklist:
-            if regex.search(string):
-                yield regex.pattern
+            for match in regex.findall(string):
+                yield match
```
Comment (JoshuaRLi) [inline detect_secrets/plugins/base.py]: This is the desired behavior.

File: detect_secrets/plugins/private_key.py [hunk 2]
```diff
@@ -1,68 +1,25 @@
-"""
-This code was extracted in part from
```
Comment (KevinHock) [inline detect_secrets/plugins/private_key.py]: I'd recommend we keep this, better safe than sorry IMO https://github.com/Yelp/detect-secrets/issues/31

Comment (JoshuaRLi) [reply detect_secrets/plugins/private_key.py]: Alright.

**Pull request merged**